types/types.go

```diff
@@ -94,6 +94,7 @@ type ImageReference interface {
 type BlobInfo struct {
 	Digest      digest.Digest // "" if unknown.
 	Size        int64         // -1 if unknown
+	Config      bool          // true if the blob is a config blob
```

Comment:
This is not very elegant... conceptually, the information is already available in `MediaType`, and more importantly if `Config:true` is only set by `Image.ConfigInfo()`, the caller structurally knows that the blob is a config without having to ask.

Maybe we should have separate `ImageDestination.{PutLayer,PutBlob}`, or at the very least least move this `Config` bool into a separate parameter of `PutBlob` which can be set by the caller directly instead of relying on `ConfigInfo`.  (@runcom WDYT?)

Reply:
…separate `imageDestination.{PutLayer,PutConfig}` that is.

Reply:
If this is currently the only need to separate layers from the config, I think it would be less intrusive to add a bool paramater to the putBlob method.
Do you think there are other use cases that need this distinction?